src/Adapter/MSTestAdapter.PlatformServices/Execution/TestMethodInfo.cs

```diff
@@ -394,6 +392,18 @@ private async Task<TestResult> ExecuteInternalAsync(object?[]? arguments, Cancel
         {
             try
             {
+                // We invoke global test initialize methods before creating the test class instance.
+                // We consider the test class constructor as a "local" initialization.
+                // We want to invoke first the global initializations, then local ones, then test method.
+                // After that, we invoke local cleanups (including Dispose) and finally global cleanups at last.
+                foreach ((MethodInfo method, TimeoutInfo? timeoutInfo) in Parent.Parent.GlobalTestInitializations)
+                {
+                    InvokeGlobalInitializeMethod(method, ref executionContext, timeoutInfo, timeoutTokenSource);
```

Comment:
TODO: I need to double check what happens in case of exception in global initialize, how is it impacting the running of the global cleanup etc.